### PR TITLE
convert sbcs codec and some tests

### PR DESCRIPTION
**Pull request by gyzerok**
Here is the update for SBCS.

My only problem is - I am not sure how to convert `sbcs-test.js` because it uses `iconv` and I am not sure what to do with some usages of `Buffer` there. Maybe it'd be better for you to take a look at this suite?

Otherwise I think everything is there.

### CODE REVIEW

**Comment by gyzerok** [comment]
The test failures seem to be from the stuff in master and are not related to my PR

File: encodings/sbcs-codec.js [hunk 1]
```diff
@@ -1,72 +1,77 @@
...
-    // Strings are immutable in JS -> we use ucs2 buffer to speed up computations.
-    var decodeBuf = this.decodeBuf;
-    var newBuf = Buffer.alloc(buf.length*2);
-    var idx1 = 0, idx2 = 0;
-    for (var i = 0; i < buf.length; i++) {
-        idx1 = buf[i]*2; idx2 = i*2;
-        newBuf[idx2] = decodeBuf[idx1];
-        newBuf[idx2+1] = decodeBuf[idx1+1];
+        for (let i = 0; i < str.length; i++)
+            bytes[i] = this.encodeBuf[str.charCodeAt(i)];
```
Comment (ashtuchkin) [inline encodings/sbcs-codec.js:51]: love how concise that is

**Comment by ashtuchkin** [comment]
Looks like the master is green now, so could you rebase and I'll merge tomorrow.

File: encodings/sbcs-codec.js (hunk 1 above)
Comment (gyzerok) [reply encodings/sbcs-codec.js:51]: The funny part is, I was expecting the the whole migration will be more like a manual labor. Just changing calls to `Buffer.from` and `Buffer.toString` to our custom functions. So it took me some time to figure out how change the code. But in the end was interesting to dig a little bit deeper into encodings business.

**Comment by gyzerok** [comment]
@ashtuchkin rebased on master. All green now.

**Comment by ashtuchkin** [comment]
Thank you!
Note I've added eslint/prettier integration on the latest master, so be sure to rebase if you decide to convert something else.

**Comment by gyzerok** [comment]
@ashtuchkin this is great news! I was missing prettier in the workflow very much :)

Which codec do you think would be better for me to take? Probably in my case better == simpler to migrate.

**Comment by ashtuchkin** [comment]
Utf32 or dbcs would be next I guess. Both will require some diving into, so
I'd understand if you wouldn't want to spend more time than you planned on
it.

**Pull request merged**